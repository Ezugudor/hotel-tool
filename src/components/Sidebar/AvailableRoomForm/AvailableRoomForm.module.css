.Container{
  
}
.FormBody{
    margin-bottom:50px;
}

.FieldCont{
    display:flex;
    align-items:center;
    justify-content:space-between;
    margin-bottom:20px;
    background:rgba(0,0,0,0.1);
    padding-left:20px;
    border-radius:4px
}

.FieldCont label{
   /* font-size:15px; */
   font-weight:bold;
 }

.FieldCont input{
   border:none;
   height:50px;
   border-radius:0 4px 4px 0;
   padding:0 20px;
   font-weight:700;
   /* font-size:20px; */
}

.Container button{
    width:100%;
    padding:20px;
    border:none;
    background:#000;
    color:#fff;
    font-weight:bold;


}