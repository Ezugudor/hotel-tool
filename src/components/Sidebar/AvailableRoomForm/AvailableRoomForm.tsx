import React, { FormEvent, useContext } from "react";
import { AvailableRoomsContext } from "../../../context";
import { useInput } from "../../../hooks";
interface AvailableRoomFormProps {
    onSubmit?: Function
}

export const AvailableRoomForm: React.FunctionComponent<AvailableRoomFormProps> = ({ onSubmit }) => {

    const economyField = useInput();
    const premiumField = useInput();
    const [, setAvailableRooms] = useContext(AvailableRoomsContext);

    const updateAvailableRoom = (e: FormEvent) => {
        e.preventDefault();
        const newAvailableRooms = { economy: parseInt(`${economyField.value}`), premium: parseInt(`${premiumField.value}`) };
        setAvailableRooms(newAvailableRooms);
        onSubmit?.(e);
    }

    return (
        <form onSubmit={(e) => { updateAvailableRoom(e) }}>
            <div>
                <div>
                    <label htmlFor="economyRoomField">Economy :</label>
                    <input type="text" name="economy" id="economyRoomField" {...economyField} />
                </div>

                <div>
                    <label htmlFor="premiumRoomField">Premium :</label>
                    <input type="text" name="economy" id="premiumRoomField" {...premiumField} />
                </div>

                <button type="submit">Enter</button>
            </div>
        </form>
    )
}