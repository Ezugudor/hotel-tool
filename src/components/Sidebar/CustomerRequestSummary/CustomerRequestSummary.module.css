.Container{
  
}

.RequestInfoHead{
    
}

.RequestInfoCont{
    border-bottom:1px solid #956cc4;
    padding:10px
}

.RequestInfo{
    display:flex;
    justify-content:space-between;
    margin-bottom:5px;
}

.RILabel{

}

.RIValue{
    font-weight:bold;
    color:#fff;
    font-size:20px
}

.RIValueCount{
    color:#0ee89c
}