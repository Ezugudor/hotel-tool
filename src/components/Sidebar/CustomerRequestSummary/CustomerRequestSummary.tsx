import React, { useContext } from "react";
import { CustomerContext } from "../../../context";
import { countRequests, sumRequests } from "../../../helpers/Helpers";
import { RoomCategory } from "../../../models/RoomCategory";
import styles from "./CustomerRequestSummary.module.css"

export const CustomerRequestSummary: React.FunctionComponent = () => {

    const customerBids = useContext(CustomerContext);


    return (
        <div className={styles.Container}>
            <div className={styles.RequestInfoCont}>
                <h4 className={styles.RequestInfoHead}>Total Requests</h4>
                <div className={styles.RequestInfoBody}>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Count: </span>
                        <span className={`${styles.RIValue} ${styles.RIValueCount}`}> {countRequests(customerBids)} </span>
                    </div>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Amount: </span>
                        <span className={styles.RIValue}> € {sumRequests(customerBids)} </span>
                    </div>
                </div>
            </div>

            <div className={styles.RequestInfoCont}>
                <h4 className={styles.RequestInfoHead}>Premium Requests</h4>
                <div className={styles.RequestInfoBody}>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Count: </span>
                        <span className={`${styles.RIValue} ${styles.RIValueCount}`}> {countRequests(customerBids, RoomCategory.PREMIUM)} </span>
                    </div>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Amount: </span>
                        <span className={styles.RIValue}> € {sumRequests(customerBids, RoomCategory.PREMIUM)} </span>
                    </div>
                </div>
            </div>

            <div className={styles.RequestInfoCont}>
                <h4 className={styles.RequestInfoHead}>Non-Premium Requests</h4>
                <div className={styles.RequestInfoBody}>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Count: </span>
                        <span className={`${styles.RIValue} ${styles.RIValueCount}`}> {countRequests(customerBids, RoomCategory.NON_PREMIUM)} </span>
                    </div>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Amount: </span>
                        <span className={styles.RIValue}> € {sumRequests(customerBids, RoomCategory.NON_PREMIUM)} </span>
                    </div>
                </div>
            </div>
        </div>

    )
}