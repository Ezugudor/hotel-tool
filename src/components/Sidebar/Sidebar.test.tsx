import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Sidebar } from '.';
import { CustomerRequestSummary } from './CustomerRequestSummary';
import { AvailableRoomForm } from './AvailableRoomForm';


let sideBarCont: ShallowWrapper;
let customerRequestSummaryCont: ShallowWrapper;
let availableRoomFormCont: ShallowWrapper;

beforeAll(() => {
    sideBarCont = shallow(<Sidebar />);
    customerRequestSummaryCont = sideBarCont.find(CustomerRequestSummary)
    availableRoomFormCont = sideBarCont.find(AvailableRoomForm)
});


describe("Sidebar Component Test", () => {
    it('should match with existing snapshot', () => {
        expect(sideBarCont.html()).toMatchSnapshot();
    })

    it("should render single CustomerRequestSummary component", () => {
        expect(customerRequestSummaryCont).toHaveLength(1)
    })

    it("should render single AvailableRoomForm component", () => {
        expect(availableRoomFormCont).toHaveLength(1);
    })

})
