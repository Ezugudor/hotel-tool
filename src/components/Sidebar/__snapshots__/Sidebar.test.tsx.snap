// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component Test should match with existing snapshot 1`] = `"<div><div><span class=\\"label\\">Total Requests Count: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Total Requests Amount: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Premium Requests Count: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Premium Requests Amount: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Non Premium Requests Count: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Non Premium Requests Amount: </span><span class=\\"value\\"> 0 </span></div></div><form><div><div><label for=\\"economyRoomField\\">Economy :</label><input type=\\"text\\" name=\\"economy\\" id=\\"economyRoomField\\" value=\\"\\"/></div><div><label for=\\"premiumRoomField\\">Premium :</label><input type=\\"text\\" name=\\"economy\\" id=\\"premiumRoomField\\" value=\\"\\"/></div><button type=\\"submit\\">Enter</button></div></form>"`;
